### DISCUSSION TITLE
use grant_type "code+token",then,how to refresh_token,please tell me

### DISCUSSION THREAD

**Discussion by Csirme** [ory/hydra#2840] [Q&A] [answered]
this is callback result

`http://localhost:5555/callback#scope=offline email phone profile address&code=PMehsP1B94i2Ed-_FdAWNsY8eI_chumU7CrRQnCP7WA.9OMOnlDn372uoQlCQs4b5rxZGVetgv-RTJPUaSAhWnU&access_token=lfADMGZZLEaaCnhFc4xiJME5M38pbJeTAZTh-zHa0r4.t1CzTJLOTmsdGPbl-ZFasV-u66YfPqWWRVb54WTt9hM&expires_in=3599&token_type=bearer&state=cftojebzoajwqmnkkoxhosga`

My question：after the access_token  expire,how to refresh a new token,not rediret to login page. (use Vue.js)

:) love ory hydra team

**Comment by vinckr** [comment, marked as answer]
Hello @Csirme 
Apologies for the late answer. 

Can you give us some more pointers or even share a repo with your frontend with me?

If the user initiates a OAuth2 Authorize Code or Implicit Flow and they don't have an active session (in your application) they will be redirected to the Login. 
Check out this documentation: https://www.ory.sh/hydra/docs/concepts/login#login-sessions-prompt-max_age-id_token_hint. You should be able to accept the login flow in the background in your frontend.

This might miss your question, would be good to know what you are trying to achieve!

**Reply by vinckr** [reply]
Hey @Csirme , just checking in if your problem still persists. 
I will mark this as answered for now, but feel free to come back to the discussion any time.
Thanks a ton :bee:
